dotnet/src/Connectors/Connectors.GoogleVertexAI.UnitTests/Core/Gemini/Clients/GeminiChatGenerationFunctionCallingTests.cs

```diff
@@ -0,0 +1,403 @@
+        chatHistory.Add(new GeminiChatMessageContent(AuthorRole.Assistant, string.Empty, "modelId", [functionCallPart]));
+        chatHistory.Add(new GeminiChatMessageContent(AuthorRole.Tool, string.Empty, "modelId", toolCallResponse));
+        var executionSettings = new GeminiPromptExecutionSettings
+        {
+            ToolCallBehavior = ToolCallBehavior.EnableFunctions([this._timePluginDate, this._timePluginNow])
+        };
+
+        // Act
+        await client.GenerateChatMessageAsync(chatHistory, executionSettings: executionSettings, kernel: this._kernelWithFunctions);
+
+        // Assert
+        GeminiRequest request = JsonSerializer.Deserialize<GeminiRequest>(this._messageHandlerStub.RequestContent)!;
+        var content = request.Contents.LastOrDefault();
+        Assert.NotNull(content);
+        Assert.Equal(AuthorRole.Tool, content.Role);
+        var functionResponse = content.Parts![0].FunctionResponse;
+        Assert.NotNull(functionResponse);
+        Assert.Equal(toolCallResponse.FullyQualifiedName, functionResponse.FunctionName);
+        Assert.Equal(JsonSerializer.Serialize(toolCallResponse.FunctionResult.GetValue<object>()), functionResponse.Response.Arguments.ToJsonString());
+    }
+
+    [Fact]
+    public async Task ShouldReturnFunctionsCalledByModelAsync()
+    {
+        // Arrange
+        this._messageHandlerStub.ResponseToReturn.Content = new StringContent(this._responseContentWithFunction);
+        var client = this.CreateChatCompletionClient();
+        var chatHistory = CreateSampleChatHistory();
+        var executionSettings = new GeminiPromptExecutionSettings
+        {
+            ToolCallBehavior = ToolCallBehavior.EnableFunctions([this._timePluginDate, this._timePluginNow])
+        };
+
+        // Act
+        var chatMessageContents =
+            await client.GenerateChatMessageAsync(chatHistory, executionSettings: executionSettings, kernel: this._kernelWithFunctions);
+
+        // Assert
+        var message = chatMessageContents.SingleOrDefault() as GeminiChatMessageContent;
+        Assert.NotNull(message);
+        Assert.NotNull(message.ToolCalls);
+        Assert.Single(message.ToolCalls,
+            item => item.FullyQualifiedName == this._timePluginNow.FullyQualifiedName);
+        Assert.Single(message.ToolCalls,
+            item => item.Arguments!["param1"]!.ToString()!.Equals("hello", StringComparison.Ordinal));
+    }
+
+    [Fact]
+    public async Task IfAutoInvokeShouldAddFunctionsCalledByModelToChatHistoryAsync()
+    {
+        // Arrange
+        using var handlerStub = new MultipleHttpMessageHandlerStub();
+        handlerStub.AddJsonResponse(this._responseContentWithFunction);
+        handlerStub.AddJsonResponse(this._responseContent);
+#pragma warning disable CA2000
+        var client = this.CreateChatCompletionClient(httpClient: handlerStub.CreateHttpClient());
+#pragma warning restore CA2000
```

Comment:
nit: Noticed this warning suppression in a couple of places. Can we do the following?
```csharp
using var httpClient = handlerStub.CreateHttpClient();
var client = this.CreateChatCompletionClient(httpClient: httpClient);
```

Reply:
If I remove pragram, pipieline will fail with formatter errors

Reply:
ChatCompletion is common and refers to Generating at once and Streaming. ChatGeneration tests only generation , ChatStreaming only streaming.